.github/workflows/test-endpoint-copier-operator-helm-chart.yml

```diff
@@ -0,0 +1,153 @@
+        
+        mkdir helm-charts
+        tar xvzf "${archive}" -C helm-charts/
+
+    - name: Install K3s
+      run: |
+        curl -sfL https://get.k3s.io | INSTALL_K3S_EXEC="server --cluster-init --write-kubeconfig-mode=644" K3S_TOKEN=foobar sh -
+
+    - name: Install Helm
+      run: |
+        curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+        chmod +x get_helm.sh
+        ./get_helm.sh
+        
+    - name: Deploy endpoint-copier-operator Helm chart
+      run: |
+        export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+        helm install endpoint-copier-operator helm-charts/endpoint-copier-operator
+        
+    - name: Wait for all endpoint-copier-operator pods to become ready
+      run: |
+        export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+        
+        # Set a timeout of 5 minutes
+        timeout=$((SECONDS + 300))
+        
+        while [ $SECONDS -lt $timeout ]; do
+            # Run the kubectl command to get pod information
+            kubectl_output=$(kubectl -n default get po | tail -n +2)
+        
+            # Flag to track whether all pods are ready
+            all_pods_ready=true
+        
+            # Iterate over each line in the kubectl output
+            while IFS= read -r line; do
+                # Extract the pod name and the readiness status
+                pod_name=$(echo "$line" | awk '{print $1}')
+                readiness_status=$(echo "$line" | awk '{print $2}')
+        
+                # Extract the desired and running replicas from the readiness status
+                desired_replicas=$(echo "$readiness_status" | awk -F'/' '{print $1}')
+                running_replicas=$(echo "$readiness_status" | awk -F'/' '{print $2}')
+        
+                # Check if the digit before / is the same as the one after /
+                if [ "$desired_replicas" -eq "$running_replicas" ]; then
+                    echo "$pod_name is ready"
+                else
+                    echo "$pod_name is not ready"
+                    all_pods_ready=false
+                fi
+            done <<< "$kubectl_output"
+        
+            # Check if all pods are ready
+            if [ "$all_pods_ready" = true ]; then
+                echo "All pods are ready"
+                exit 0
+            fi
+        
+            # Wait for a moment before checking again
+            sleep 5
+        done
+        
+        # If the loop completes, it means the timeout occurred
+        echo "Timeout: Not all pods are ready in 5 minutes."
+        exit 1
+
+    - name: Define Kubernetes VIP service
+      run: |
+        export KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+
+        cat <<-EOF | kubectl apply -f -
+        apiVersion: v1
+        kind: Service
+        metadata:
+          name: kubernetes-vip
+          namespace: default
+          labels:
+            servicetype: kubernetes-vip
+        spec:
+          ports:
+          - name: rke2-api
```

Comment:
This will fail as cannot be two ports with the same name. I suggest to keep this one as rke2-api but the other one rename it as k8s-api

Reply:
Also in this case, k3s doesn't need 9345, it is just rke2

Reply:
They do not have the same name actually (rke2-api and , rke2-api2)

Reply:
> Also in this case, k3s doesn't need 9345, it is just rke2
The purpose of the test is to test if the endpoint-copier-operator will create the endpoint with two ports successfully.

The test just needs to create a Service with two random ports and see what will happen with the endpoint

Reply:
I will change the name of the ports to port1 and port2 just to not making any confusions

Reply:
sorry, I misread it.